llvm/test/CodeGen/RISCV/O3-pipeline.ll

```diff
@@ -68,6 +68,9 @@
 ; CHECK-NEXT:       Expand reduction intrinsics
 ; CHECK-NEXT:       Natural Loop Information
 ; CHECK-NEXT:       TLS Variable Hoist
+; CHECK-NEXT:       Scalar Evolution Analysis
+; CHECK-NEXT:       Loop Pass Manager
+; CHECK-NEXT:         EVL-based Induction Variables Simplify
```

Comment:
Is possible to get this into the same loop pass manager as LSR?

Reply:
IIUC in order to do that we need to put EVLIndVarSimplify side-by-side with LSR, but I feel like we can make good uses of canonical form loops even between LSR and CodeGenPrepare. Do you think that will be the case?

Reply:
I'm not sure. LSR itself already does some things that make loops harder to analyze. Though it doesn't make them not countable.